doc/source/serve/autoscaling.md

```diff
@@ -0,0 +1,43 @@
+(serve-autoscaling)=
+
+# Serve Autoscaling
+
+This section should help you:
+
+- Understand how Ray Serve autoscaling works
+- Learn how to configure the parameters for your workload
+
+
+## Autoscaling architecture
+Ray Serve autoscaling feature automatically increases or decreases the number of replicas based on the load. 
+
+![pic](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling.svg)
+
+- Deployment handle(Client) and worker replica periodically push the stats to the autoscaler.
+- Autoscaler requires serve handle queue metrics and replicas requests metrics to make decision whether to scale (up or down) the replicas.
+- Deployment Handle(Client) keeps polling the replica stats from controller to get the updated replicas information. Serve Handle(Client) will send requests directly to the replica based on the replicas information (Round Robin).
+
+:::{note}
+When the controller dies, the client is still able to send requests, but autoscaling will pause to work. When the controller recovers, the autoscaling will start working, all previous metrics collected will be lost.
+:::
+
+## Autoscaling parameters
+There are several parameters the autoscaling algorithm takes into consideration when deciding the target replicas for your deployment
+**min_replicas**: The minimal number of replicas for the deployment, the min_replicas will also be the initial number replicas when the deployment is deployed.
+:::{note}
+Ray Serve Autoscaling allows the `min_replicas` to be 0 to start your deployment, the scale up will be started when you start sending traffic. There will be cold start time during the period, ray serve handle will wait (block) for available replicas to assign the request.
+:::
+**max_replicas**: Max replicas is the maximum number of replicas for the deployment. Ray Serve Autoscaling will rely on the Ray Autoscaler to scale up more nodes when the resource is not enough to bring up more replicas. 
+**target_num_ongoing_requests_per_replica**: The config is to maintain how many ongoing requests concurrently running per replica at most. If the number is lower, the scale up will be done more aggressively.
```

Comment:
Does `target_num_ongoing_requests_per_replica` set a hard limit on the number of requests a deployment can handle at a certain number of replicas, until a scale-up happens? The phrasing of "at most" makes it seem like that's the case.

Reply:
Actually, it is more about the replica view. 
For deployment perspective: 
the number of requests = (running requests + pending requests) * num of replicas.

Reply:
I second Shreyas's concern, is `target_num_ongoing_requests_per_replica` an upper bound?  From what I remember it's just a target (the true value might be higher or lower), so "at most" might be a little misleading.

Reply:
I see the point, let me rephrase. Thank you!